packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/boolean/component.rs

```diff
@@ -0,0 +1,86 @@
+use crate::components::prelude::*;
+use crate::general_state_var::{BooleanStateVar, BooleanToStringStateVar, StateVarAlias};
+
+/// Definition of the `<boolean>` DoenetML component
+#[derive(
+    Debug,
+    Default,
+    ComponentNode,
+    ComponentState,
+    ComponentActions,
+    ComponentAttributes,
+    RenderedChildren,
+)]
+#[no_rendered_children]
+pub struct Boolean {
+    /// The common component data needed to derive the `ComponentNode` trait
+    pub common: ComponentCommonData,
+
+    /// The state variables that underlie the `<boolean>` component.
+    pub state: BooleanState,
+}
+
+impl Boolean {
+    /// If a Boolean component extends a state variable of type boolean,
+    /// then its value state variable should shadow that variable.
+    ///
+    /// Returns: a tuple of (component type, state variable index)
+    pub fn get_state_variable_that_shadows_when_extending() -> (&'static str, StateVarIdx) {
+        (
+            Boolean::get_component_type(),
+            BooleanState::get_value_state_variable_index(),
+        )
+    }
+}
+
+/// The state variables that underlie the `<boolean>` component.
+#[derive(Debug, ComponentState)]
+pub struct BooleanState {
+    /// The value of the `<boolean>` component.
+    ///
+    /// It is marked `is_public` so that it can be referenced in DoenetML via `.value`.
+    ///
+    /// It is marked `for_renderer` to send this value to the renderer of the `<boolean>` component.
+    ///
+    /// It is marked with a component profile state variable, indicating that the `<boolean>` component
+    /// can represent a boolean value by returning the value of this state variable.
+    #[is_public]
+    #[for_renderer]
+    #[component_profile_state_variable]
+    value: StateVar<bool>,
+
+    /// An alias to the `value` state variable.
+    ///
+    /// It is marked `is_public` so that it can be referenced in DoenetML via `.boolean`.
```

Comment:
_Somewhere_, we want clear documentation of what `#[is_public]` means, but we don't need to repeat it everywhere.

Reply:
Yeah, just not sure where the *somewhere* is. We don't yet have a place where we are documenting how state works in general. Maybe in the definition of the `ComponentState` trait?